### ISSUE TITLE
panelmode PROBLEM Warning: Valid 4 digit PIN not found

### ISSUE THREAD

**Issue by jonshorter** [davesmeghead/visonic#135]
Morning Dave

I've not posted before, so I will start by thanking you for such a thorough and professional implementation.

I have a Powermaster 10 with a Wemos and everything works great except for persistent repeating errors in the logs (see below). If I remove the default code from the panel config interface, and force keypad, the code is accepted. Also, with no code in the UI I can instant arm fine (but not disarm). With a code in the UI I can disarm and instant arm, but the log fills with this error and it shows Triggered flashing in the UI after a while, but that doesn't seem to impact any functionality. 

Is there a config file I need to check? I've done a lot of research on here and the forums, but don't understand the logic of override codes and saw somewhere that some config codes sometimes don't/didn't used to get cleared properly - is this a route I could explore?

I have uninstalled the integration, ESPHome and deleted the custom-integration visonic folder before reinstalling, but no dice.

`"2024-08-08 01:46:51.934132  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM",
      "2024-08-08 01:48:42.069180  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM",
      "2024-08-08 01:50:17.189038  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM",
      "2024-08-08 01:50:27.199707  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM",
      "2024-08-08 01:52:17.357241  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM",
      "2024-08-08 01:52:27.369176  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM",
      "2024-08-08 01:52:37.382893  W P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM"` etc....

A pointer in the right direction would be greatly appreciated!

Thanks.

**Comment by davesmeghead** [comment]
Hi and thankyou,
Can you tell me the version of the integration that you're using please?
If you are not using 0.9.6.20 then please update the software and try again.
If you are using 0.9.6.20 (or you update to it) and have the problem then please set the logger settings to debug for this integration, zip the logfile and upload it here in a reply.   Instructions [here](https://github.com/davesmeghead/visonic/wiki#logger-settings) for the logger settings.
Thanks
Dave

**Comment by jonshorter** [comment]
Hi Dave

I took some logs tonight before double-checking the version (I was SURE i had updated to .20! Must have been before the reinstall though.) - the interface was showing .13 and however, and when I went into settings, there was an update to .20 from .14 waiting for me!

So, I updated to .20 and ran a second full batch of tests using each permutation of settings. All seems good now, and the logs are much cleaner - the work you've done over the past week has really improved things a lot, when I compare the 2 logs.

I have noticed that if I try to Arm Away when there's an Internal-F type zone activated (e.g. door open) then the error reported is that the code has been rejected by the panel (which is true, I guess, though not through an incorrect code).

In earlier versions, I believe I was getting repeated "P0  Warning: Valid 4 digit PIN not found, panelmode is PROBLEM" due to a prior event where ARM AWAY had failed due to a door being open, but then that that was never clearing properly and so there were perpetual log entries as per my initial post. Here's what the logs looked like in .13 FYI... the 'Siren Sounding' must have been what was causing the false Triggered status in HA. 

2024-08-09 23:27:39.026 DEBUG (SyncWorker_60) [custom_components.visonic.client] P0  Send command to Visonic Alarm Panel: ARM_AWAY
2024-08-09 23:27:39.027 DEBUG (SyncWorker_60) [custom_components.visonic.client] P0  Client: Sending HA Event visonic_alarm_command_to_panel  {'panel': 0, 'panel_id': 'alarm_control_panel.visonic_alarm', 'action': 'armdisarm', 'ready': False, 'open': ['binary_sensor.visonic_z07'], 'bypass': [], 'tamper': [], 'zonetamper': [], 'command': 'Arm_Away', 'reason': 0, 'reason_str': 'Success', 'message': 'Request Arm/Disarm Success, sent Command to Panel'}
2024-08-09 23:27:39.568 DEBUG (MainThread) [custom_components.visonic.pyvisonic] [sendPdu] Sending Command ((Dis)Arming System)    waiting for message response ['0X2']
2024-08-09 23:27:39.782 DEBUG (MainThread) [custom_components.visonic.pyvisonic] [handle_msgtype08] Access Denied  len 1 data 43
2024-08-09 23:27:39.782 DEBUG (MainThread) [custom_components.visonic.pyvisonic] [handle_msgtype08]                last command a1 00 00 05 78 43 07 00 00 00 00 43
2024-08-09 23:27:39.782 DEBUG (MainThread) [custom_components.visonic.pyvisonic] [handle_msgtype08] Attempt to send a command message to the panel that has been denied, wrong pin code used
2024-08-09 23:27:39.786 DEBUG (MainThread) [custom_components.visonic.client] P0  Client: Sending HA Event visonic_alarm_panel_state  {'panel': 0, 'panel_id': 'alarm_control_panel.visonic_alarm', 'action': 'pinrejected'}
2024-08-09 23:27:39.787 DEBUG (MainThread) [custom_components.visonic.client] P0  notification_config ['siren_sounding']
2024-08-09 23:27:39.787 INFO (MainThread) [custom_components.visonic.client]  P0  HA Warning (not shown in frontend due to user config), condition is AvailableNotifications.INVALID_PIN message=The Pin Code has been Rejected By the Panel

Anyway, seems like there's nothing to see here and we can all move on :) Thanks for everything.
